### PR TITLE
feat: support environment file for extensions

### PR DESCRIPTION
**Pull request by frezbo**
Supports setting `environmentFile` for Talos System Extension Services.

Fixes: #7316

### CODE REVIEW

**Comment by frezbo** [comment]
Tested with the tailscale extension:
tailscale

```bash
❯ talosctl -n 10.5.0.3 get links 
NODE       NAMESPACE   TYPE         ID                VERSION   TYPE       KIND     HW ADDR                                           OPER STATE   LINK STATE
10.5.0.3   network     LinkStatus   bond0             1         ether      bond     aa:01:84:18:4c:16                                 down         false
10.5.0.3   network     LinkStatus   dummy0            1         ether      dummy    7a:6e:05:f7:17:63                                 down         false
10.5.0.3   network     LinkStatus   enx966a24fb42ad   4         ether               96:6a:24:fb:42:ad                                 up           true
10.5.0.3   network     LinkStatus   flannel.1         3         ether      vxlan    02:6a:25:4c:79:e3                                 unknown      true
10.5.0.3   network     LinkStatus   ip6tnl0           1         tunnel6    ip6tnl   00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00   down         false
10.5.0.3   network     LinkStatus   lo                2         loopback            00:00:00:00:00:00                                 unknown      true
10.5.0.3   network     LinkStatus   sit0              1         sit        sit      00:00:00:00                                       down         false
10.5.0.3   network     LinkStatus   tailscale0        4         nohdr      tun                                                        unknown      true
10.5.0.3   network     LinkStatus   teql0             1         void                                                                  down         false
10.5.0.3   network     LinkStatus   tunl0             1         ipip       ipip     00:00:00:00                                       down         false
```

```bash
❯ talosctl -n 10.5.0.3 get addresses
NODE       NAMESPACE   TYPE            ID                                                       VERSION   ADDRESS                                       LINK
10.5.0.3   network     AddressStatus   enx966a24fb42ad/10.5.0.3/24                              1         10.5.0.3/24                                   enx966a24fb42ad
10.5.0.3   network     AddressStatus   enx966a24fb42ad/fe80::946a:24ff:fefb:42ad/64             2         fe80::946a:24ff:fefb:42ad/64                  enx966a24fb42ad
10.5.0.3   network     AddressStatus   flannel.1/10.244.1.0/32                                  1         10.244.1.0/32                                 flannel.1
10.5.0.3   network     AddressStatus   flannel.1/fe80::6a:25ff:fe4c:79e3/64                     2         fe80::6a:25ff:fe4c:79e3/64                    flannel.1
10.5.0.3   network     AddressStatus   lo/127.0.0.1/8                                           1         127.0.0.1/8                                   lo
10.5.0.3   network     AddressStatus   lo/::1/128                                               1         ::1/128                                       lo
10.5.0.3   network     AddressStatus   tailscale0/100.105.xx.x/32                               1         100.105.xx.x/32                               tailscale0
10.5.0.3   network     AddressStatus   tailscale0/fd7a:115c:a1e0:ab12:4843:cd96:6269:4f02/128   1         fd7a:115c:a1e0:ab12:4843:cd96:6269:4f02/128   tailscale0
10.5.0.3   network     AddressStatus   tailscale0/fe80::3818:cf1b:a917:dbb3/64                  1         fe80::3818:cf1b:a917:dbb3/64                  tailscale0
```

And if the environment file is absent, the extensions waits:

```bash
[   10.189243] [talos] service[ext-tailscale](Waiting): Waiting for file "/var/etc/tailscale/auth.env" to exist
```

File: internal/app/machined/pkg/system/services/extension.go [hunk 1]
```diff
@@ -64,6 +67,13 @@ func (svc *Extension) PostFunc(r runtime.Runtime, state events.ServiceState) (er
 func (svc *Extension) Condition(r runtime.Runtime) conditions.Condition {
 	conds := []conditions.Condition{}
 
+	if svc.Spec.Container.EnvironmentFile != "" {
+		// add a dependency on the environment file
+		svc.Spec.Depends = append(svc.Spec.Depends, extservices.Dependency{
```
Comment (smira) [inline internal/app/machined/pkg/system/services/extension.go:72]: this mutates the extension spec, and potentially multiple times, I think it's much better to build directly a `conditions.WaitForFileToExist` here

File: internal/app/machined/pkg/system/services/extension.go [hunk 2]
```diff
@@ -189,3 +203,31 @@ func (svc *Extension) APIStartAllowed(runtime.Runtime) bool {
 func (svc *Extension) APIStopAllowed(runtime.Runtime) bool {
 	return true
 }
+
+func (svc *Extension) parseEnvironment() ([]string, error) {
+	var envVars []string
+
+	if svc.Spec.Container.EnvironmentFile != "" {
+		envFile, err := os.OpenFile(svc.Spec.Container.EnvironmentFile, os.O_RDONLY, 0)
```
Comment (smira) [inline internal/app/machined/pkg/system/services/extension.go:211]: file is not closed?

File: internal/app/machined/pkg/system/services/extension.go (hunk 1 above)
Comment (frezbo) [reply internal/app/machined/pkg/system/services/extension.go]: fixed

File: internal/app/machined/pkg/system/services/extension.go (hunk 2 above)
Comment (frezbo) [reply internal/app/machined/pkg/system/services/extension.go:209]: fixed

**Review by smira — APPROVED** [review]

**Pull request merged**